src/controllers/main/main.ts

```diff
@@ -603,17 +601,15 @@ export class MainController extends EventEmitter {
   async updateSelectedAccount(selectedAccount: string | null = null, forceUpdate: boolean = false) {
     if (!selectedAccount) return
 
-    this.portfolio.updateSelectedAccount(
-      this.accounts,
-      this.settings.networks,
-      selectedAccount,
-      undefined,
-      { forceUpdate }
-    )
-
-    const account = this.accounts.find(({ addr }) => addr === selectedAccount)
-    if (shouldGetAdditionalPortfolio(account))
-      this.portfolio.getAdditionalPortfolio(selectedAccount)
+    this.portfolio
+      .updateSelectedAccount(this.accounts, this.settings.networks, selectedAccount, undefined, {
+        forceUpdate
+      })
+      .then(() => {
+        const account = this.accounts.find(({ addr }) => addr === selectedAccount)
+        if (shouldGetAdditionalPortfolio(account))
+          this.portfolio.getAdditionalPortfolio(selectedAccount)
+      })
```

Comment:
I don't see the reason this being in `.then()`. It could be the way it was before without problems?